ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -97,7 +100,10 @@ internal constructor(
   private var shouldLoadFromDraft: Boolean = savedStateHandle[TASK_SHOULD_LOAD_FROM_DRAFT] ?: false
   private var draftDeltas: List<ValueDelta>? = null
 
-  private val activeSurvey: Survey = requireNotNull(surveyRepository.activeSurvey)
+  private val activeSurvey: Survey = runBlocking {
+    surveyRepository.activeSurveyFlow.filterNotNull().first()
+  }
```

Comment:
What will happen if the coroutine is never completed? Will the app freeze until ANR error occurs?

Reply:
During the standup today, @anandwana001 suggested adding a timeout here. @gino-m I suggested that we allow the timeout exception to propagate and crash the app, since not being able to load the survey from the local db would be an unrecoverable error. 

Ideally, we would ensure the survey was active *before* rendering the `DataCollectionFragment`. We could then use `SurveyRepository.activeSurvey` directly rather than the Flow to get the current survey. @shobhitagarwal1612 Is this possible, based on how the fragment gets activated?

Reply:
> Ideally, we would ensure the survey was active before rendering the DataCollectionFragment. We could then use SurveyRepository.activeSurvey directly rather than the Flow to get the current survey. @shobhitagarwal1612 Is this possible, based on how the fragment gets activated?

I don't think so, the fragment gets restored when the activity is started and the injected dependencies are initialized  after the fragment is created.

Reply:
> I don't think so, the fragment gets restored when the activity is started and the injected dependencies are initialized after the fragment is created.

On configuration change, yes, but on app restart we explicitly redirect to this fragment if there's an unsaved draft, no? On configuration change the VM should still exist, so this wouldn't be an issue. In `HomeScreenFragment.onViewCreated`, however, we navigate to the data collection screen if there's an unsaved draft.

That said, I agree it would be more robust for the fragment which needs the survey to handle the edge cases, rather than the navigation source. 

@anandwana001  Let's keep the fix in this PR plus timeout as a workaround, we can add other handle delayed load and other edge cases if it turns out to be an issue.